libretro/libretro.c

```diff
@@ -508,94 +512,70 @@ void update_variables()
 {
     struct retro_variable var;
 
-    var.key = "mupen64plus-rspmode";
+    var.key = CORE_NAME "-rspmode";
     var.value = NULL;
     if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
     {
-        if (!strcmp(var.value, "HLE"))
-            rspMode = 0;
-        else
-            rspMode = 1;
+        rspMode = strcmp(var.value, "HLE") ? 1 : 0;
```

Comment:
Honestly not sure if I like these, for a experienced programmer its no Issue, but it can be confusing for others.
It's fine with me tho.

Reply:
I think we should level programmers up and not down :) The conditional operator is a very elegant tool IMHO and programmers should be more aware of them.

I agree that in this particular case `strcmp()` can be a bit confusing because it has an "inverse" logical meaning, when the strings are equal you get `0` ("false") instead of `!=0` ("true"). Perhaps I can add back the negation operator (`!`) so it reads more naturally:

    rspMode = !strcmp(var.value, "HLE") ? 0 : 1;

what do you think?

Reply:
I like this more, and yes I have this particular thought because of what you cited.

Reply:
sure then, I will add the negations so they read more naturally :)

Reply:
Now that I was doing this little refactoring, I have a question.
In this code block (as is currently):
```c
var.key = CORE_NAME "-txFilterIgnoreBG";
if (!strcmp(var.value, "True"))
    txFilterIgnoreBG = 0;
else
    txFilterIgnoreBG = 1;
```
Shouldn't it be the other way around? True=Ignore, False=Don't ignore? is this correct?

Reply:
I answer myself, the option is defined as this:
```
      { "mupen64plus-txFilterIgnoreBG",
            "Filter background textures; True|False" },
```
Meaning that indeed True=Filter (don't ignore), False=DontFilter (do ignore)
Was a bit confusing :D